packages/backend/src/models/AIConfig.ts

```diff
@@ -55,14 +59,25 @@ export function parseYaml(raw: string, defaultArch: string): AIConfig {
 
   return {
     application: {
-      containers: containers.map(container => ({
-        arch: Array.isArray(container['arch']) ? container['arch'] : [defaultArch],
-        modelService: container['model-service'] === true,
-        containerfile: isString(container['containerfile']) ? container['containerfile'] : undefined,
-        contextdir: assertString(container['contextdir']),
-        name: assertString(container['name']),
-        gpu_env: Array.isArray(container['gpu-env']) ? container['gpu-env'] : [],
-      })),
+      containers: containers.map(container => {
+        if (typeof container !== 'object') throw new Error('containers array malformed');
+
+        let contextdir: string;
+        if ('contextdir' in container) {
+          contextdir = assertString(container['contextdir']);
+        } else {
+          contextdir = path.basename(path.dirname(filepath));
```

Comment:
I think it will fail. If there is no contextdir specified, should it be `.` or empty?
let's say the file path is `/root/folder/ai-studio.yaml`, i guess the resulting `contextdir` would be `folder` right? But what we do is (below) so it will result in `/root/folder/folder` and will fail
https://github.com/projectatomic/ai-studio/blob/4b8b7892f083e433b20df702e7445588965963c1/packages/backend/src/managers/applicationManager.ts#L353-L354

Reply:
Hummmm yeah you are right, there are some repetitive logic then, I will remove the one I added in this PR